frontend/lib/src/components/elements/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement, ReactNode, useEffect, useState } from "react"
+
+import { SIZE } from "baseui/modal"
+
+import Modal, {
+  ModalHeader,
+  ModalBody,
+} from "@streamlit/lib/src/components/shared/Modal"
+import { notNullOrUndefined } from "@streamlit/lib/src/util/utils"
+import { Block as BlockProto } from "@streamlit/lib/src/proto"
+
+export interface Props {
+  element: BlockProto.Dialog
+  children?: ReactNode
```

Comment:
nit: I'm not sure if this is correct to add `children` here as part of the props. Maybe something we need to double check and eventually also fix in some other layout containers.

Reply:
oh yeah, I think you are right! Also, removing it did not break anything 😁